### PR TITLE
Sync Atomikos changes with master

### LINKED ISSUE #25451: Sync master with Atomikos changes and Send PR
$Subject.

### PR DESCRIPTION
**Pull request by gimantha**
Fixes #25451

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/build.gradle [hunk 1]
```diff
@@ -26,6 +26,9 @@ dependencies {
     implementation 'io.opentracing:opentracing-util'
     implementation 'org.awaitility:awaitility'
     implementation 'org.apache.commons:commons-lang3'
+    implementation('com.atomikos:transactions-jta:3.6.4') {
+        exclude group: 'org.hibernate', module: 'hibernate'
+    }
```
Comment (mohanvive) [inline bvm/ballerina-runtime/build.gradle:31]: @gimantha Any reason to use 10 years old Atomikos version? Shall we bump to latest version, WDYT?

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/transactions/TransactionResourceManager.java [hunk 2]
```diff
@@ -181,13 +176,21 @@ public boolean notifyCommit(Strand strand, String transactionId, String transact
...
         List<BallerinaTransactionContext> txContextList = resourceRegistry.get(combinedId);
         if (txContextList != null) {
+            Transaction trx = trxRegistry.get(combinedId);
+            try {
+                if (trx != null) {
+                    trx.commit();
+                }
+
+            } catch (SystemException | HeuristicMixedException | HeuristicRollbackException | RollbackException e) {
+                log.error("error in commit the transaction, " + combinedId + ":" + e.getMessage(), e);
```
Comment (mohanvive) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/transactions/TransactionResourceManager.java:186]: I think we need rephrase the error message here?

Comment (mohanvive) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/transactions/TransactionResourceManager.java:186]: Same rephrasing required for abort transaction.

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/transactions/TransactionResourceManager.java [hunk 3]
```diff
@@ -321,8 +335,6 @@ private void invokeCommittedFunction(Strand strand, String transactionId, String
             for (int i = fpValueList.size(); i > 0; i--) {
                 FPValue fp = fpValueList.get(i - 1);
                 //TODO: Replace fp.getFunction().apply
-//                BRuntime.getCurrentRuntime().invokeFunctionPointerAsyncIteratively(fp, 1, () -> args,
-//                        results -> {}, () -> null);
                 fp.getFunction().apply(args);
```
Comment (mohanvive) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/transactions/TransactionResourceManager.java:338]: Haven't we got fix from Runtime team for this issue.

File: misc/testerina/modules/testerina-core/src/main/ballerina/Ballerina.toml [hunk 4]
```diff
@@ -8,5 +8,5 @@ target = "java8"
     [[platform.libraries]]
         artifactId = "mock"
         version = "0.0.0"
-        path = "./lib/testerina-core-2.0.0-Preview4.jar"
+        path = "./lib/testerina-core-2.0.0-Preview5-SNAPSHOT.jar"
```
Comment (mohanvive) [inline misc/testerina/modules/testerina-core/src/main/ballerina/Ballerina.toml:11]: Do we need this change?

File: bvm/ballerina-runtime/build.gradle (hunk 1 above)
Comment (gimantha) [reply bvm/ballerina-runtime/build.gradle:31]: updated

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/transactions/TransactionResourceManager.java (hunk 3 above)
Comment (gimantha) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/transactions/TransactionResourceManager.java:338]: created an issue for this
https://github.com/ballerina-platform/ballerina-lang/issues/26187

**Pull request closed without merging**